lib/idempo.rb

```diff
@@ -55,6 +55,9 @@ def call(env)
       end
 
       status, headers, body = @app.call(env)
+      # `body` could be of type `ActionDispatch::Response::RackBody` and idempo will not even attempt to store it,
+      # we're converting ActionDispatch::Response::RackBody to a storable array format.
+      body = body.try(:to_ary) || body
```

Comment:
The reason why the behavior differs is likely because there was some tweaking of the rack body in Rails due to the changes in Rack 3.x. The issue with calling `to_ary` explicitly is that it instantly "materializes" the entire Rack body into strings, and `to_ary` comes from the Rack 3.x protocol (and has Semantics™ attached to it). There are implications to this (look for RackBody changes in Rails and the related PR discussions and https://github.com/appsignal/appsignal-ruby/pull/1037/files#diff-542053b5df2402727a27d180e5f6f63bf8a646b81d72fa0c013de4177a6becafR23 ).

So it is not as much as needing to "convert" something into something else as it is about knowing how large the response is going to be once it gets read out of the body. Can you check why https://github.com/julik/idempo/blob/main/lib/idempo.rb#L118 does not work? Doesn't Rails set content length by itself?

Reply:
@julik I actually traced it to a next line, that rejects to save this body -- since body is not an Array.
https://github.com/julik/idempo/blob/main/lib/idempo.rb#L120

Maybe `return false unless body.is_a?(Array) || body.respond_to(:to_ary)`?

Reply:
yes but do `response_headers` contain a `Content-Length` of some description in your testing? I would expect Rails sets it but I'm not sure